sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -343,6 +349,28 @@ public String getSchemaResourceType(String schema, String resourceName)
     return plannerToolbox.rootSchema().getResourceType(schema, resourceName);
   }
 
+  /**
+   * Add a lookup name to load in the lookup loading spec.
+   */
+  public void addLookupToLoad(String lookupName)
+  {
+    if (lookupLoadingSpec.getLookupsToLoad() == null) {
+      lookupLoadingSpec = LookupLoadingSpec.loadOnly(Collections.singleton(lookupName));
+    } else {
+      Set<String> existingLookupsToLoad = new HashSet<>(lookupLoadingSpec.getLookupsToLoad());
+      existingLookupsToLoad.add(lookupName);
+      lookupLoadingSpec = LookupLoadingSpec.loadOnly(existingLookupsToLoad);
+    }
+  }
+
+  /**
+   * Returns the lookup loading spec for a given task.
```

Comment:
```suggestion
   * Lookup loading spec for MSQ tasks.
```

Reply:
It's not limited to MSQ tasks though? Right now we consume it only in MSQ, but there's no restriction to not use this for other tasks in future.